src/helm/joanie/templates/app/job_db_migrate.yaml

```diff
@@ -0,0 +1,47 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "joanie.fullname" . }}-dbmigrate
+  namespace: {{ .Release.Namespace | quote }}
+  labels: {{- include "joanie.labels" . | nindent 4 }}
+    {{- if .Values.commonLabels }}
+    {{ toYaml .Values.commonLabels | nindent 4 }}
+    {{- end }}
+    app.kubernetes.io/component: joanie
+  {{- if .Values.commonAnnotations }}
+  annotations: {{ toYaml .Values.commonAnnotations | nindent 4 }}
```

Comment:
I suggest that you add Helm Hook for Jobs, the purpose is to avoid the deploy of your apps before the end of your migration job, and a potential failure.

A hook can be like this:
```  
annotations:
    "helm.sh/hook": pre-upgrade, pre-install
    "helm.sh/hook-delete-policy": before-hook-creation
    "helm.sh/hook-delete-timeout": "0"
    "helm.sh/hook-weight": "5"
```

So when you install/upgrade the chart, only the jobs is deployed and executed, when it success, the other ressources will be deployed, you can add multiple hook and order them with the `hook-weight` annotation

Reply:
You can still use `"helm.sh/hook-weight": "5"` to order the execution of the hook, you start by adding the hook on the postgres's svc and endpoint. Please note that hook weights can be positive or negative numbers but must be represented as strings. When Helm starts the execution cycle of hooks of a particular Kind it will sort those hooks in ascending order.

Reply:
yes , I think that is the best way to implement this. Thank you!

Reply:
another idea, in some case you don't need to run the db-migration, for eg. a minor release, or a bug, you don't want to wait extra time just for your db-migrate to run for nothing, in this case, you may add a feature flag, a bool var `run_db_migrate = false` and in your template add `{{- if .Values.run_db_migrate }}` in that case you have the control on when you want to start the migration and you may gain some deploy time